adapters/httpapi_winhttp.c

```diff
@@ -820,6 +944,49 @@ HTTPAPI_RESULT HTTPAPI_CloneOption(const char* optionName, const void* value, co
                 result = HTTPAPI_OK;
             }
         }
+        else if (strcmp(OPTION_HTTP_PROXY, optionName) == 0)
+        {
+            *savedValue = (HTTP_PROXY_OPTIONS*)malloc(sizeof(HTTP_PROXY_OPTIONS));
+            if (*savedValue == NULL)
+            {
+                LogError("failure in malloc");
+                result = HTTPAPI_ERROR;
+            }
+            else
+            {
+                HTTP_PROXY_OPTIONS *savedOption = (HTTP_PROXY_OPTIONS*) *savedValue;
+                HTTP_PROXY_OPTIONS *option = (HTTP_PROXY_OPTIONS*) value;
+                if (mallocAndStrcpy_s((char**)(savedOption->host_address),
+                                      (const char*)option->host_address) != 0)
+                {
+                  LogError("unable to clone the proxy host adress content");
```

Comment:
If any failures occur you need to release any memory allocated within this function and NULL the *savedValue variable

Reply:
This applies to each block bellow as well.